examples/Catena4430_Sensor/Catena4430_cMeasurementLoop_SDcard.cpp

```diff
@@ -571,4 +575,75 @@ cMeasurementLoop::updateFromSd(
         }
     }
 
+void
+cMeasurementLoop::handleSdTTNv3Migrate(
+    void
+    )
+    {
+    static const char * const sMigrate = "MIGRATE.V3";
+    bool fMigrate;
+    bool fResult = this->checkSdCard();
+
+	if (fResult)
+        {
+        fMigrate = this->handleSdTTNv3MigrateCardUp(sMigrate);
+        }
+
+    if (fMigrate)
+        {
+        bool fFramUpdate = this->updateFramAppEui();
+
+        if (fFramUpdate)
+            {
+            this->reJoinNetwork();
+            gSD.remove(sMigrate);
+            gLog.printf(gLog.kInfo, "cFramStorage::kAppEUI: update: success\n");
+			}
+        else
+            gLog.printf(gLog.kError, "cFramStorage::kAppEUI: not updated\n");
+        }
+    }
+
+bool
+cMeasurementLoop::handleSdTTNv3MigrateCardUp(
+    const char *sMigrate
+    )
+    {
+    if (! gSD.exists(sMigrate))
+        {
+        if (gLog.isEnabled(gLog.kTrace))
+            gLog.printf(gLog.kAlways, "%s: not found: %s\n", FUNCTION, sMigrate);
+
+        return false;
+        }
+    return true;
+    }
+
+bool
+cMeasurementLoop::updateFramAppEui(
+    void
+    )
+    {
+    auto const pFram = gCatena.getFram();
+    uint8_t AppEUI[] = { 1, 0, 0, 0, 0, 0, 0, 0 };
```

Comment:
change to `static const`. Otherwise ther will be runtime initialization, wasteful.